test/backend_gather.in.cpp

```diff
@@ -36,6 +36,78 @@ using namespace ngraph;
 
 static string s_manifest = "${MANIFEST}";
 
+NGRAPH_TEST(${BACKEND_NAME}, gather_4d_indices_no_axis)
+{
+    Shape params_shape{3, 2};
+    Shape indices_shape{2, 3, 4};
+    Shape out_shape{2, 3, 4, 2};
+    auto P = make_shared<op::Parameter>(element::f32, params_shape);
+    auto I = make_shared<op::Parameter>(element::i32, indices_shape);
+    auto G = make_shared<op::Gather>(P, I);
+    auto f = make_shared<Function>(make_shared<op::GetOutputElement>(G, 0), ParameterVector{P, I});
+
+    auto backend = runtime::Backend::create("${BACKEND_NAME}");
+
+    // Create some tensors for input/output
+    auto p = backend->create_tensor(element::f32, params_shape);
+    copy_data(p, vector<float>{1.0f, 1.1f, 2.0f, 2.1f, 3.0f, 3.1f});
+    auto i = backend->create_tensor(element::i32, indices_shape);
+    copy_data(
+        i, vector<int32_t>{0, 1, 1, 2, 0, 1, 1, 2, 0, 1, 1, 2, 0, 1, 1, 2, 0, 1, 1, 2, 0, 1, 1, 2});
+    auto result = backend->create_tensor(element::f32, out_shape);
+
+    auto c = backend->compile(f);
+    c->call_with_validate({result}, {p, i});
+    EXPECT_TRUE(test::all_close_f(
+        (vector<float>{1.0f, 1.1f, 2.0f, 2.1f, 2.0f, 2.1f, 3.0f, 3.1f, 1.0f, 1.1f, 2.0f, 2.1f,
```

Comment:
does it always copy the whole row? based on the kernel code, i see you are calculating offset and extent. What indices would you need so that you don't copy the complete row?

Reply:
When axis is 0 and the params has rank r,  gather always copies slices of rank r-1. In this unit test, the shape of params is {3, 2}. The offset for params slice would be {i, 0} and the extent would be {1, 3} with i be 0, 1, or 2.